Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveHM_2nd.cpp

```diff
@@ -0,0 +1,902 @@
+                            Hz_eff += MacroscopicProperties::face_avg_to_face(i, j, k, 0, amrex::IntVect(0, 0, 1), amrex::IntVect(0, 1, 0), Hz);
+                        }
+
+                        // calculate the a_temp_static_coeff (it is divided by 2.0 because the input dt is actually dt/2.0)
+                        Real a_temp_dynamic_coeff = PhysConst::mu0 * amrex::Math::abs(mag_gamma_arry) / 2.0;
+
+                        // calculate a_temp_yface
+                        // x component on y-faces of grid
+                        a_temp_yface(i, j, k, 0) = (M_normalization != 0) ? -(dt * a_temp_dynamic_coeff * Hx_eff + a_temp_static_yface(i, j, k, 0))
+                                                                          : -(dt * a_temp_dynamic_coeff * Hx_eff + 0.5 * a_temp_static_yface(i, j, k, 0) 
+                                                                            + 0.5 * mag_alpha_arry * 1. / std::sqrt(std::pow(M_yface(i, j, k, 0), 2.0) + std::pow(M_yface(i, j, k, 1), 2.0) + std::pow(M_yface(i, j, k, 2), 2.0)) * M_old_yface(i, j, k, 0));
+
+                        // y component on y-faces of grid
+                        a_temp_yface(i, j, k, 1) = (M_normalization != 0) ? -(dt * a_temp_dynamic_coeff * Hy_eff + a_temp_static_yface(i, j, k, 1))
+                                                                          : -(dt * a_temp_dynamic_coeff * Hy_eff + 0.5 * a_temp_static_yface(i, j, k, 1) 
+                                                                            + 0.5 * mag_alpha_arry * 1. / std::sqrt(std::pow(M_yface(i, j, k, 0), 2.0) + std::pow(M_yface(i, j, k, 1), 2.0) + std::pow(M_yface(i, j, k, 2), 2.0)) * M_old_yface(i, j, k, 1));
+
+                        // z component on y-faces of grid
+                        a_temp_yface(i, j, k, 2) = (M_normalization != 0) ? -(dt * a_temp_dynamic_coeff * Hz_eff + a_temp_static_yface(i, j, k, 2))
+                                                                          : -(dt * a_temp_dynamic_coeff * Hz_eff + 0.5 * a_temp_static_yface(i, j, k, 2) 
+                                                                            + 0.5 * mag_alpha_arry * 1. / std::sqrt(std::pow(M_yface(i, j, k, 0), 2.0) + std::pow(M_yface(i, j, k, 1), 2.0) + std::pow(M_yface(i, j, k, 2), 2.0)) * M_old_yface(i, j, k, 2));
+
+                        // update M_yface from a and b using the updateM_field
+                        // x component on y-faces of grid
+                        M_yface(i, j, k, 0) = MacroscopicProperties::updateM_field(i, j, k, 0, a_temp_yface, b_temp_static_yface);
+
+                        // y component on y-faces of grid
+                        M_yface(i, j, k, 1) = MacroscopicProperties::updateM_field(i, j, k, 1, a_temp_yface, b_temp_static_yface);
+
+                        // z component on y-faces of grid
+                        M_yface(i, j, k, 2) = MacroscopicProperties::updateM_field(i, j, k, 2, a_temp_yface, b_temp_static_yface);
```

Comment:
```suggestion
```

Reply:
this is added in the icomp for loop above

Reply:
Don't accept this (comment above)